public/jobs.js

```diff
@@ -0,0 +1,352 @@
+          }
+          suspendInput = false;
+        } else {
+          // this is an update
+          suspendInput = true;
+          try {
+            const jobID = editJob.dataset.id;
+            const response = await fetch(`/api/v1/jobs/${jobID}`, {
+              method: "PATCH",
+              headers: {
+                "Content-Type": "application/json",
+                Authorization: `Bearer ${token}`,
+              },
+              body: JSON.stringify({
+                company: company.value,
+                position: position.value,
+                status: status.value,
+              }),
+            });
+            const data = await response.json();
+            if (response.status === 200) {
+              message.textContent = "The entry was updated.";
+              showing.style.display = "none";
+              company.value = "";
+              position.value = "";
+              status.value = "pending";
+              thisEvent = new Event("startDisplay");
+              document.dispatchEvent(thisEvent);
+            } else {
+              message.textContent = data.msg;
+            }
+          } catch (err) {
+  
+            message.textContent = "A communication error occurred.";
+          }
+        }
+        suspendInput = false;
+      } // section 5
+      else if (e.target.classList.contains("editButton")) {
+        editJob.dataset.id = e.target.dataset.id;
+        suspendInput = true;
+        try {
+          const response = await fetch(`/api/v1/jobs/${e.target.dataset.id}`, {
+            method: "GET",
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: `Bearer ${token}`,
+            },
+          });
+          const data = await response.json();
+          if (response.status === 200) {
+            company.value = data.job.company;
+            position.value = data.job.position;
+            status.value = data.job.status;
+            showing.style.display = "none";
+            showing = editJob;
+            showing.style.display = "block";
+            addingJob.textContent = "update";
+            message.textContent = "";
+          } else {
+            // might happen if the list has been updated since last display
+            message.textContent = "The jobs entry was not found";
+            thisEvent = new Event("startDisplay");
+            document.dispatchEvent(thisEvent);
+          }
+        } catch (err) {
+          message.textContent = "A communications error has occurred.";
+        }
+        suspendInput = false;
+      }
+      //this is a delete
+      else if(e.target.classList.contains("deleteButton")) {
+        editJob.dataset.id = e.target.dataset.id;
+        suspendInput = true;
+        try {
+            const response = await fetch(`/api/v1/jobs/${e.target.dataset.id}`, {
+              method: "DELETE",
+              headers: {
+                "Content-Type": "application.json",
```

Comment:
Delete Request Header Typo: In the delete request for deleting a job entry, you have a typo in the Content-Type header. It should be "Content-Type": "application/json" instead of "Content-Type": "application.json".